### PR TITLE
Resourcify

### PR DESCRIPTION
**Pull request by Brightergy-SamarAcharya**
@Brightergy/kc can you please review this? This is the basic skeleton for the upcoming API client/SDK. There are some places that might require cleanup but just wanted to get initial working crap.

### CODE REVIEW

File: lib/brighterx/api.ex [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    params = opts[:params] || %{}
+    token = System.get_env("JWT")
+    req_header = request_header(%{token: token})
+    module
+    |> build_url(id, params)
+    |> Api.get(req_header)
+    |> Parser.parse(module)
+  end
+
+  def get_company(name) when is_bitstring(name), do: get("companies", %{name: name})
```
Comment (Brightergy-Bruce) [inline lib/brighterx/api.ex]: Should these be rewritten to use "find"?

File: lib/brighterx/api.ex [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  - For all other errors, a tuple of {:ok, <error_detail_map>, status_code} is returned
+  """
+  def find(module, opts \\ []) do
+    id = opts[:id] || nil
+    params = opts[:params] || %{}
+    token = System.get_env("JWT")
+    req_header = request_header(%{token: token})
+    module
+    |> build_url(id, params)
+    |> Api.get(req_header)
```
Comment (Brightergy-Bruce) [inline lib/brighterx/api.ex]: I am assuming this is using HTTPoison.Base's method?

Comment (Brightergy-SamarAcharya) [inline lib/brighterx/api.ex]: @Brightergy-Bruce yeah it is using the HTTPoison.Base's method.

File: lib/brighterx/api.ex (hunk 1 above)
Comment (Brightergy-SamarAcharya) [inline lib/brighterx/api.ex]: @Brightergy-Bruce yeah that's what I'm going to do soon.

**Pull request merged**